src/Microsoft.PowerShell.Commands.Utility/commands/utility/UnblockFile.cs

```diff
@@ -119,18 +120,13 @@ protected override void ProcessRecord()
                     {
                         AlternateDataStreamUtilities.DeleteFileStream(path, "Zone.Identifier");
                     }
-                    catch (Win32Exception accessException)
+                    catch (IOException)
```

Comment:
`File.Delete` ignores `File_Not_Found` error and throw exception for all other errors, see: http://source.dot.net/#System.IO.FileSystem/System/IO/FileSystem.Windows.cs,30dc9ae494d8f0a5

So, we should use `catch (Exception e)` here and write a non-terminating error. The error ID `RemoveItemUnauthorizedAccessError` and `ErrorCategory.PermissionDenied` may not be appropriate, we need a more general error.

Reply:
Seems the more common message doesn't make sense - the only exception useful for user that we really can get is `UnauthorizedAccessException`. The rest may be silently ignored https://msdn.microsoft.com/en-us/library/system.io.ioexception(v=vs.110).aspx because they're a temporary. We could just don't catch its at all or insert `Assert`.

Reply:
As @dantraMSFT wrote above catching all exception is not good practice and it is better catch only exceptions for which we can make behavior better. 
I want to understand how we should do the best way. Now everywhere in our code we catch all exceptions and write customized errors or re-throw. It looks very costly.

Here only `UnauthorizedAccessException` is of interest. The remaining exceptions are edge cases and we could pass them.
Of course we can catch all exceptions, write a generic message and put the original exception in inner exception but it can mask a problem.

Reply:
If you don't catch all exceptions in this case, any uncaught exceptions will stop the cmdlet run and all following paths won't be processed. That would be a breaking change.

Whether catching all exception is right or wrong depends on the scenario. In this instance, we don't want the failure of processing one path to block processing other files.

Reply:
Thanks for your clarification!

Fixed.

Reply:
Thanks.
I think what @dantraMSFT means is that we should not blindly catch all exceptions. As long as we know why we should catch all exception, it's OK to use `catch (Exception)`.